### PR TITLE
Increase contrast between active and inactive tabs

### PR DESCRIPTION
**Pull request by simonhong**
Resolves https://github.com/brave/brave-browser/issues/8576

Screen Shot 2020-10-20 at 9 53 53 PM
Screen Shot 2020-10-20 at 9 54 12 PM
Screen Shot 2020-10-20 at 9 55 44 PM

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: patches/chrome-browser-ui-views-tabs-tab_style_views.cc.patch [hunk 1]
```diff
@@ -1,16 +1,48 @@
 diff --git a/chrome/browser/ui/views/tabs/tab_style_views.cc b/chrome/browser/ui/views/tabs/tab_style_views.cc
-index 950c967b5f87a6e3f531e3155a420002a7510d3a..a9b5b74a1a5c345727c1deb631fce7fcbb2762dc 100644
+index 950c967b5f87a6e3f531e3155a420002a7510d3a..ad84681855bc39f55796c8740f490b998e2bc68a 100644
```
Comment (simonhong) [inline patches/chrome-browser-ui-views-tabs-tab_style_views.cc.patch:2]: Can't avoid patching. `GM2TabStyle` is defined in anonymous namespace.

**Review by karenkliu — APPROVED** [review]
Looks great! 💯

**Comment by mkarolin** [comment]
Do tab separators seem a bit lost now? @karenkliu 
TabColors

File: chromium_src/chrome/browser/ui/views/frame/browser_view.cc [hunk 2]
```diff
@@ -4,10 +4,14 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "brave/browser/ui/views/tabs/brave_browser_tab_strip_controller.h"
+#include "brave/browser/ui/views/tabs/brave_tab_strip.h"
 #include "brave/browser/ui/views/toolbar/brave_toolbar_view.h"
+#include "chrome/browser/ui/views/frame/tab_strip_region_view.h"
```
Comment (mkarolin) [inline chromium_src/chrome/browser/ui/views/frame/browser_view.cc:9]: Is this needed?

**Comment by karenkliu** [comment]
@mkarolin Yep - this is the intended design. The favicon + title of inactive tabs already do a lot to differentiate between tabs, so the tab separator contrast was reduced. This will also help the active tab stand out more within the tabbar.

File: patches/chrome-browser-ui-views-tabs-tab_style_views.cc.patch (hunk 1 above)
Comment (mkarolin) [reply patches/chrome-browser-ui-views-tabs-tab_style_views.cc.patch]: @simonhong: wasn't sure this was going to work, so had to try it out and it seems to be working. What do you think of subclassing instead like this: https://github.com/brave/brave-core/commit/4c19f3e0674dc0e460e5a0885fc659fe8d3a4aac

File: chromium_src/chrome/browser/ui/views/frame/browser_view.cc (hunk 2 above)
Comment (simonhong) [reply chromium_src/chrome/browser/ui/views/frame/browser_view.cc:9]: Yup, tab_strip_region_view.h references `TabStrip`.
W/o this, I got undefind linke error.
```
Undefined symbols for architecture x86_64:
  "TabStripRegionView::TabStripRegionView(std::__Cr::unique_ptr<BraveTabStrip, std::__Cr::default_delete<BraveTabStrip> >)", referenced from:
      BrowserView::BrowserView(std::__Cr::unique_ptr<Browser, std::__Cr::default_delete<Browser> >) in libui.a(browser_view.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

File: patches/chrome-browser-ui-views-tabs-tab_style_views.cc.patch (hunk 1 above)
Comment (simonhong) [reply patches/chrome-browser-ui-views-tabs-tab_style_views.cc.patch]: Thanks @mkarolin ! Overriding `CreateForTab` is great :)

**Review by mkarolin — APPROVED** [review]
LGTM

File: chromium_src/chrome/browser/ui/views/frame/browser_view.cc (hunk 2 above)
Comment (mkarolin) [reply chromium_src/chrome/browser/ui/views/frame/browser_view.cc:9]: Ah, makes sense!

**Comment by simonhong** [comment]
@mkarolin Thanks for great review 👍

**Pull request merged**